### ISSUE TITLE
Make contact sort and search match the mockup

### ISSUE THREAD

**Issue by ProMcTagonist** [qTox/qTox#2008]
I've gotten nothing but negative reactions about this:

![sort](https://cloud.githubusercontent.com/assets/4358079/8719046/a00b83f2-2b5d-11e5-972b-1741c9929291.png)

Also, it would help https://github.com/tux3/qTox/issues/1237#issuecomment-88569902 be done cleanly, like this:

![mockup](https://cloud.githubusercontent.com/assets/7714365/6947314/569756ba-d874-11e4-85c4-47d717897943.PNG)

Maybe the sort button could drop into a text box in addition to options like the friend request button drops into pending requests. It could autofocus the search field to retain one-click access.

**Comment by TheSpiritXIII** [comment]
What happens to the "Online" combo box when you have no friend requests? You have a giant empty space?

**Comment by zetok** [comment]
@ProMcTagonist how exactly mock-up defines searching for contacts? From what it seems, it doesn't…

**Comment by ProMcTagonist** [comment]
@zetok I've made a proposition just under the picture but it would be nice if @itsduke himself had an idea to share.

**Comment by ItsDuke** [comment]
@TheSpiritXIII yes, you would have empty space. Why would it show someone if no one is online? I'd rather it tell me the truth instead of prepopulating my friend list with offline people whom I can't talk to.

@zetok as was always stated for the mockup, it's a dropdown and you select from a set of filters: online, recent, all, and search. When you select search, the box expands and you are free to type within it, and it will display all contacts relevant to your search.

**Comment by suhr** [comment]
> When you select search, the box expands and you are free to type within it, and it will display all contacts relevant to your search.

Sounds reasonable. But it also should start searching on typing, without selecting anything.

**Comment by zetok** [comment]
@ItsDuke right. Sadly, this doesn't sound good enough - i.e. I'm forced to unnecessarily click around just to be able to search - only after that I'm apparently supposed to be able to search for contacts?

I.e. too much hassle to actually use this functionality, since you'd have to e.g. switch from `All` → `Search` just to be able to search for contacts…

**Comment by TheSpiritXIII** [comment]
@ItsDuke Something else what would be nice to have in addition to the friend requests are group invite requests. This is something that your mockups don't have and if we're going to replace the friend request dialogs we may as well replace the invite request dialogs as well.

**Comment by ProMcTagonist** [comment]
"# Friend Request(s)" or "# Group Invite(s)" or "# Requests" for both sounds workable.

**Comment by ItsDuke** [comment]
@suhr Something doesn't sit right with me about the "typing randomly=search", but perhaps I'm misunderstanding.

@zetok You're not "forced" to click around. You make it sound like it's this arduous journey, when it's just a way to filter your friends list, and is placed logically with the other ways to filter your list. It's literally two clicks, or hell, even just one if you hold the mouse down to bring up the dropdown, and then release the mouse when you get to "Search". How is two (or one) clicks too much hassle?

As I said previously, these mockups are not a hard and fast rule for building Tox, you're free to disregard them completely. I have a feeling you won't be satisfied until Tox automatically begins searching for you based on your thought patterns, though :wink: 

@TheSpiritXIII & @ProMcTagonist Exactly. It could just say requests, or invites, or some other text altogether.
